### ISSUE TITLE
storage: leaked error: via acceptance test: storage/stores.go:87: store 3 not found

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#5166]
The following test appears to have failed:

[#14363](https://circleci.com/gh/cockroachdb/cockroach/14363):

```
W160311 16:04:01.203602 acceptance/chaos_test.go:251  round 2: cluster recovered
I160311 16:04:01.582964 acceptance/chaos_test.go:179  client 0 shutting down
I160311 16:04:01.600609 acceptance/chaos_test.go:179  client 1 shutting down
I160311 16:04:01.649956 acceptance/chaos_test.go:362  21 3.4/sec
I160311 16:04:01.750345 acceptance/cluster/localcluster.go:607  stopping
--- FAIL: TestClusterRecovery (14.47s)
	chaos_test.go:269: pq: storage/stores.go:87: store 3 not found
=== RUN   TestNodeRestart
I160311 16:04:04.671486 acceptance/cluster/localcluster.go:270  Initializing Cluster:
Cluster Setup: AdHoc 3x1
Duration: 5s, Stall: 2m0s
Node 0 - Store 0
Node 1 - Store 0
Node 2 - Store 0
Total Nodes Count:3, Total Store Count:3
I160311 16:04:04.671670 acceptance/cluster/docker.go:73  ImagePull cockroachdb/builder:20160305-182433 starting
--
--- SKIP: TestBuildBabyCluster (0.00s)
	terraform_test.go:30: only enabled during testing
=== RUN   TestFiveNodesAndWriters
--- SKIP: TestFiveNodesAndWriters (0.00s)
	util_test.go:64: running in docker mode
FAIL
ok github.com/cockroachdb/cockroach/acceptance 1337s

```
Please assign, take a look and update the issue accordingly.

**Comment by tschottdorf** [comment]
This error comes from `stores.go`:
```
		return nil, util.Errorf("store %d not found", storeID)
```

note that it's not a structured error, so no retries are possible. If a node is just restarted, probably it is asked to handle requests before it has initialized all its stores. With that possibility, the right thing to do is to convert this into a DistSender-retryable error (maybe a RangeKeyMismatchError).

**Comment by tamird** [comment]
Seems likely to be fallout from d9cf809047bdb170e7f5c7ea260384d11aba5cb9

On Fri, Mar 11, 2016 at 9:45 PM, Tobias Schottdorf <notifications@github.com

**Comment by tschottdorf** [comment]
the offending API calls are mostly `Start`, but sometimes `Kill`, and I think I remember seeing `Wait`.

**Comment by tamird** [comment]
I think this was fixed by #5441.
